synapse/util/threepids.py

```diff
@@ -48,3 +48,25 @@ def check_3pid_allowed(hs, medium, address):
         return True
 
     return False
+
+
+def check_3pid_valid_format(medium, address):
+    """Checks whether 3pid has a valid format
+
+    Args:
+        medium (str): 3pid medium - e.g. email, msisdn
+        address (str): 3pid address to check (e.g. "wotan@matrix.org")
```

Comment:
Thanks for add a docstring! We've been slowly moving toward using Python 3 type hints, please move the types here to the function definition, e.g.: 

```python
def check_3pid_valid_format(medium: str,  address: str) -> bool:
```